zipline/extensions.py

```diff
@@ -0,0 +1,44 @@
+import re
+
+
+class ExtensionArgs(object):
```

Comment:
imo we should not have the parsing code and consuming code in the same place. If a user is looking to see which extension arguments have been set, right now they will see `extension_args`, `parse_extension`, and `get_namespace` in addition to whatever was set. This makes it harder for us to maintain because we will be making this part of the public interface. Instead, we could make the root group an `Arg` or `Namespace`, and in the `main`, parse into this structure and that make this empty object available as `zipline.extension_args`.